server/repository/users.go

```diff
@@ -1,34 +1,65 @@
 package repository
 
 import (
+	"database/sql"
 	"fmt"
 
 	"github.com/src-d/code-annotation/server/model"
 )
 
-// Users is the User repository
+// Users repository
 type Users struct {
-	users []*model.User
+	db *sql.DB
 }
 
-// Create stores a user into the DB
-func (r *Users) Create(m *model.User) error {
-	for _, u := range r.users {
-		if u.ID == m.ID {
-			return nil
-		}
+// NewUsers returns a new Users repository
+func NewUsers(db *sql.DB) *Users {
+	return &Users{db: db}
+}
+
+// Create stores a User into the DB, and returns that new User
+func (repo *Users) Create(user *model.User) error {
+
+	_, err := repo.db.Exec(
+		"INSERT INTO users (login, username, avatar_url, role) VALUES ($1, $2, $3, $4)",
+		user.Login, user.Username, user.AvatarURL, user.Role)
+
+	if err != nil {
```

Comment:
_idem_
Can the insertion succeed, but the `Get` fail?
```go
return err
```

Reply:
Even if it's unlikely, the Get may fail if the DB connection breaks, or someone deletes the entry at that very exact moment.
In that case the user pointer is not correctly populated with the assigned ID, and I'd say the method should notify of this returning an error.

Reply:
*PS:*
if it is assumed a possible race condition and the necessity of calling `.Get` from the `.Create` to modify the `user` retrieved as an input parameter, it can happen unexpected things;
It can happen that the final (modified) `user` will be quite different than the input one (another Name, Login...) so the caller function will operate with that `user` without noticing it was "changed in behind" by the `.Create` func.

(that's another reason because it could be a bad idea to modify the input user inside the `.Create` func)

Reply:
I'm okay with current solution and your proposal is okay too. Both approaches are common. For error in `get` and not in `exec` people invented transactions if we really need it.

I would suggest to merge it as is now.